indexer/ai/src/main/java/com/google/cloudsearch/ai/BaseAISkill.java

```diff
@@ -0,0 +1,164 @@
+        }
+        else {
+            setAISkillName(aiSkillName);
+        }
+    }
+
+    /**
+     *
+     * @param outputMappingList     List of output mappings
+     */
+    @Override
+    public void setOutputMappings(List<OutputMapping> outputMappingList) {
+        this.outputMappings = outputMappingList;
+    }
+
+    /**
+     * @return  Returns the List of Output Mappings
+     */
+    @Override
+    public List<OutputMapping> getOutputMappings() {
+        return this.outputMappings;
+    }
+
+    /**
+     * Parse output Mapping
+     * @param outputMapping     JSON Array of Output Mappings
+     * @param schema            CloudSearch schema
+     * @throws InvalidConfigException   Throws exception if property name not present in schema
+     */
+    @Override
+    public void parseOutputMappings(JSONArray outputMapping, JSONObject schema) throws InvalidConfigException {
+        if (outputMapping == null) {
+            throw new InvalidConfigException("Output Mapping not specified.");
+        }
+        List<OutputMapping> outputMappingList = new ArrayList<OutputMapping>();
+        for(Object objMap : outputMapping) {
+            JSONObject mappingObject = (JSONObject) objMap;
+            OutputMapping obj = new OutputMapping();
+            if (isValidPropertyName((String) mappingObject.get(Constants.CONFIG_TARGET_PROPERTY), schema)) {
+                obj.setPropertyName((String) mappingObject.get(Constants.CONFIG_TARGET_PROPERTY));
+            }
+            else {
+                throw new InvalidConfigException("Invalid Object or Property Name");
+            }
+            obj.setSkillOutputField((String) mappingObject.get(Constants.CONFIG_OUTPUT_FILED));
+            outputMappingList.add(obj);
+        }
+        setOutputMappings(outputMappingList);
+    }
+
+    /**
+     * Parse the JSONObject for the skill in the Configuration.
+     * Call the setter functions and initialize all the data required for the skill.
+     * Throws InvalidConfigException if the skill config is invalid.
+     *
+     * @param aiSkill        JSONObject for the skill
+     * @param schema         CloudSearch schema
+     * @throws InvalidConfigException   If the configuration for mapping skill outputs
+     *                                  to schema objects is invalid then throws InvalidConfigException.
+     */
+    protected abstract void parse(JSONObject aiSkill, JSONObject schema) throws InvalidConfigException;
+
+    /**
+     * Build Natural Language API Document.
+     * @param inputLanguage     Language specified in Skill Configuration
+     * @param contentOrURI      Content or Cloud Storage URI
+     * @return
+     */
+    protected Document buildNLDocument(String inputLanguage, String contentOrURI) {
+        Document doc;
+        Document.Builder docBuilder = Document.newBuilder();
+
+        if(CloudStorageHandler.isCloudStorageURI(contentOrURI)) {
```

Comment:
what if my content starts with gs://

Reply:
I am using absolute system path which usually starts with /Users/... for the file names.

Reply:
I am also trying to find if there are any methods that can be used to check if a file exists

Reply:
I am assuming content is a text string